bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BallerinaXmlSerializer.java

```diff
@@ -141,8 +147,9 @@ private void writeXMLComment(XmlComment xmlValue) throws XMLStreamException {
     }
 
     private void writeXMLText(XmlText xmlValue) throws XMLStreamException {
-        // No need to escape xml text when they are within xml element, it's handle from xml stream writer.
-        if (this.withinElement) {
+        // No need to escape xml text when they are within xml element or if it is not from default stream writer.
+        // It's handled by xml stream  writer.
+        if (this.withinElement && !isDefaultFactory) {
```

Comment:
Can we use xmlOutputFactory.getProperty("escapeCharacters") != null. and equal to true here

Reply:
If we use the `xmlOutputFactory.getProperty("escapeCharacters")` in this check, it  gives an `IllegalStateException` for the non-default stream writer (Since it is not a predefined property for `WstxOutputFactory`).